File: clients/csharp/Src/elencyConfig/Authorization.cs
using ElencyConfig.Hashers;
using System;
// ReSharper disable IdentifierTypo

namespace ElencyConfig
{
    internal static class Authorization
    {
        public static string GenerateAuthorizationHeader(ElencyConfiguration config, string path, string method, bool hmac = false)
        {
            var dateTime1970 = new DateTime(1970, 1, 1);
            var span = DateTime.UtcNow - dateTime1970;
            var timestamp = span.TotalMilliseconds.ToString("0"); 
            var nonce = Guid.NewGuid().ToString();
            var value = $"{config.AppId}{path.ToLower()}{method.ToLower()}{nonce}{timestamp}";
            var key = hmac ? config.HMACAuthorizationKey : config.ConfigEncryptionKey;
            var signature = HMACSHA256.Hash(value, key);
            return $"{config.AppId}:{signature}:{nonce}:{timestamp}";
        }
    }
}

File: clients/node/lib/authorization.js
'use strict';

const hasher = require('./hashers/hmac-sha256'),
  uuid = require('uuid').v4;

function getHeaderData(config, path, method) {
  const nonce = uuid();
  const timestamp = Date.now();
  const value = `${config.appId}${path.toLowerCase()}${method.toLowerCase()}${nonce}${timestamp}`;
  return {
    nonce,
    timestamp,
    value
  };
}

module.exports = {
  generateAuthorizationHeader: async function(config, path, method, hmac) {
    hmac = hmac || false;
    const headerData = getHeaderData(config, path, method);
    const key = (hmac === true ? config.HMACAuthorizationKey : config.configEncryptionKey);
    const signature = await hasher.hash(headerData.value, key);
    return `${config.appId}:${signature}:${headerData.nonce}:${headerData.timestamp}`;
  }
};